azurerm/resource_arm_private_link_service.go

```diff
@@ -0,0 +1,649 @@
+}
+
+func expandArmPrivateLinkServiceFrontendIPConfiguration(input []interface{}) *[]network.FrontendIPConfiguration {
+	if len(input) == 0 {
+		return nil
+	}
+
+	results := make([]network.FrontendIPConfiguration, 0)
+
+	for _, item := range input {
+		result := network.FrontendIPConfiguration{
+			ID: utils.String(item.(string)),
+		}
+
+		results = append(results, result)
+	}
+
+	return &results
+}
+
+func expandArmPrivateLinkServicePrivateEndpointConnection(input []interface{}) *[]network.PrivateEndpointConnection {
+	results := make([]network.PrivateEndpointConnection, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		id := v["id"].(string)
+		privateEndpoint := v["private_endpoint"].([]interface{})
+		privateLinkServiceConnectionState := v["private_link_service_connection_state"].([]interface{})
+		name := v["name"].(string)
+
+		result := network.PrivateEndpointConnection{
+			ID:   utils.String(id),
+			Name: utils.String(name),
+			PrivateEndpointConnectionProperties: &network.PrivateEndpointConnectionProperties{
+				PrivateEndpoint:                   expandArmPrivateLinkServicePrivateEndpoint(privateEndpoint),
+				PrivateLinkServiceConnectionState: expandArmPrivateLinkServicePrivateLinkServiceConnectionState(privateLinkServiceConnectionState),
+			},
+		}
+
+		results = append(results, result)
+	}
+	return &results
+}
+
+func expandArmPrivateLinkServicePrivateLinkServicePropertiesVisibility(input []interface{}) *network.PrivateLinkServicePropertiesVisibility {
+	if len(input) == 0 {
+		return nil
+	}
+
+	subscriptions := make([]string, 0)
+
+	for _, v := range input {
+		subscriptions = append(subscriptions, v.(string))
+	}
+
+	result := network.PrivateLinkServicePropertiesVisibility{
+		Subscriptions: &subscriptions,
+	}
+
+	return &result
+}
+
+func expandArmPrivateLinkServicePrivateEndpoint(input []interface{}) *network.PrivateEndpoint {
+	if len(input) == 0 {
+		return nil
+	}
+	v := input[0].(map[string]interface{})
+
+	id := v["id"].(string)
+	location := azure.NormalizeLocation(v["location"].(string))
+	t := v["tags"].(map[string]interface{})
+
+	result := network.PrivateEndpoint{
+		ID:       utils.String(id),
+		Location: utils.String(location),
+		Tags:     tags.Expand(t),
+	}
+	return &result
+}
+
+func expandArmPrivateLinkServicePrivateLinkServiceConnectionState(input []interface{}) *network.PrivateLinkServiceConnectionState {
+	if len(input) == 0 {
+		return nil
+	}
+	v := input[0].(map[string]interface{})
+
+	status := v["status"].(string)
+	description := v["description"].(string)
+	actionRequired := v["action_required"].(string)
+
+	result := network.PrivateLinkServiceConnectionState{
+		ActionRequired: utils.String(actionRequired),
+		Description:    utils.String(description),
+		Status:         utils.String(status),
+	}
+	return &result
+}
+
+func flattenArmPrivateLinkServicePrivateLinkServicePropertiesAutoApproval(input *network.PrivateLinkServicePropertiesAutoApproval) []string {
```

Comment:
Could we remove this and use the `utils.flattenstringslice` function?

Reply:
Fixed, but I kept the function as explained above.